src/system/unix/system.c

```diff
@@ -192,6 +210,28 @@ unsigned long z_clock_elapsed_s(z_clock_t *instant) {
     return elapsed;
 }
 
+void z_clock_advance_us(z_clock_t *clock, unsigned long duration) {
+    clock->tv_sec += duration / 1000000;
+    clock->tv_nsec += (duration % 1000000) * 1000;
+
+    if (clock->tv_nsec >= 1000000000) {
+        clock->tv_sec += 1;
+        clock->tv_nsec -= 1000000000;
+    }
```

Comment:
```suggestion
    if (clock->tv_nsec >= 1000000000) {
        clock->tv_sec += clock->tv_nsec / 1000000000;
        clock->tv_nsec %= 1000000000;
    }
 ```
This will be more accurate.